docs/pages/user-guides/testing/specflow.md

```diff
@@ -0,0 +1,194 @@
+
+    // ...
+  }
+}
+```
+
+
+## Step definition methods
+
+"Step definitions" are special methods that execute Gherkin steps from feature files.
+Whenever a Gherkin scenario runs, SpecFlow "glues" each Gherkin step to its associated step definition.
+Each step definition method has an annotation with the step type (`Given`, `When`, or `Then`)
+and a regular expression to match the step text.
+With Boa Constrictor, eacn step definition should be only a few lines of Screenplay interactions, like
+`Actor.AttemptsTo(...)`, `Actor.AsksFor(...)`, and `Actor.WaitsUntil(...)`.
+
+Below is a binding class named `DuckDuckGoSteps`
+with step definitions for the three Gherkin steps from the DuckDuckGo search test.
+The constructor receives the Actor object via dependency injection from SpecFlow through scenario context.
+Each step definition then uses the Actor to call Interactions.
+The specific pages (`SearchPage` and `ResultPage`) and custom interactions (`SearchDuckDuckGo`)
+come from the [`Boa.Constrictor.Example`](https://github.com/q2ebanking/boa-constrictor/tree/main/Boa.Constrictor.Example) project.
+
+```csharp
+using Boa.Constrictor.Screenplay;
+using Boa.Constrictor.WebDriver;
+using TechTalk.SpecFlow;
+
+namespace Boa.Constrictor.Example
+{
+  [Binding]
+  public sealed class DuckDuckGoSteps : Steps
+  {
+    private IActor Actor { get; set; }
+
+    // The `InitializeScreenplay` hook from `DuckDuckGoHooks`
+    // injects the Actor object into scenario context.
+    // SpecFlow can resolve that dependency
+    // in other binding classes using constructor arguments.
+
+    public DuckDuckGoSteps(IActor actor)
+    {
+      Actor = actor;
+    }
+
+    [Given(@"the DuckDuckGo home page is displayed")]
+    public void GivenTheDuckDuckGoHomePageIsDisplayed()
+    {
+      Actor.AttemptsTo(Navigate.ToUrl(SearchPage.Url));
+    }
+
+    [When(@"the user searches for the phrase ""(.*)""")]
+    public void WhenTheUserSearchesForThePhrase(string phrase)
+    {
+      Actor.AttemptsTo(SearchDuckDuckGo.For(phrase));
+    }
+
+    [Then(@"the results page shows result links for the phrase")]
+    public void ThenTheResultsPageShowsResultLinksForThePhrase()
+    {
+      Actor.WaitsUntil(Appearance.Of(ResultPage.ResultLinks), IsEqualTo.True());
+    }
+  }
+}
+```
+
+
+## After-scenario hooks
+
+After-scenario hooks are analogous to 
+[NUnit `[TearDown]`]({{ "/user-guides/testing-with-nunit/#screenplay-teardown-methods" | relative_url }}) methods:
+they run after each test to perform cleanup operations, whether the test passed or failed.
+After-scenario hooks must be located in binding classes and bear the `[AfterScenario]` attribute.
+Like before-scenario hooks, they may or may not have an explicit order.
+
+Below is an after-scenario hook that quits the browser.
+It should be located in the `DuckDuckGoHooks` binding class:
+
+```csharp
+    [AfterScenario]
+    public void QuitBrowser()
+    {
+      Actor.AttemptsTo(QuitWebDriver.ForBrowser());
```

Comment:
If somebody copies this snippet into `DuckDuckGoHooks` as provided above the code won't compile because `Actor` isn't defined.

Reply:
You're right. More copypasta.